### ISSUE TITLE
Handle multi-Image / Scene, multi-Resolution metadata correctly

### ISSUE THREAD

**Issue by JacksonMaxfield** [tlambert03/ome-types#71]
Hey @tlambert03, got an interesting one for ya.

Even though `aicsimageio==4.0.0` isn't out yet, we have already started the planning for `4.1.0` which includes support for multi-resolution / pyramid files. We got some great test files from Seb over at Zeiss and I converted to OME-TIFF w/ `bfconvert -noflat ...`, and the resulting file can be downloaded [here](https://open.quiltdata.com/b/aics-modeling-packages-test-resources/tree/aicsimageio/test_resources/resources/variable_scene_shape_first_scene_pyramid.ome.tiff).

## Describing the file + metadata:
The file has two scenes with the first scene having multiple resolutions. In total there are four resolutions in the first scene and one resolution for the second scene. When converted, because OME doesn't encapsulate resolution data into something of it's own type, all four resolutions get converted to their own `Image` element in the metadata, resulting in the metadata having five total `Image` elements (1 scene * 4 resolutions + 1 scene * 1 resolutions).

## Now, the bug:
What is interesting is that if I crack the produced file open and parse the metadata w/ `ome-types` it produces this:
```python
In [1]: from ome_types import from_tiff

In [2]: ome = from_tiff("aicsimageio/tests/resources/variable_scene_shape_first_scene_pyramid.ome.tiff")

In [3]: ome
Out[3]: 
OME(
   creator='OME Bio-Formats 6.6.0',
   experimenters=[<1 Experimenters>],
   images=[<2 Images>],
   instruments=[<1 Instruments>],
   structured_annotations=[<5194 Structured_Annotations>],
   uuid='urn:uuid:54bab916-61f0-451d-bfbe-251896f608fb',
)
```

It correctly parses that there are two images (the two scenes regardless of multi-resolution) behavior.

**But**, if I check the metadata for the second scene:
```python
In [4]: ome.images[1].pixels
Out[4]: 
Pixels(
   id='Pixels:1',
   dimension_order='XYCZT',
   size_c=1,
   size_t=1,
   size_x=422,
   size_y=2030,
   size_z=1,
   type='uint8',
   big_endian=False,
   channels=[<1 Channels>],
   interleaved=False,
   physical_size_x=0.9082107048835328,
   physical_size_y=0.9082107048835328,
   planes=[<1 Planes>],
   significant_bits=8,
   tiff_data_blocks=[<1 Tiff_Data_Blocks>],
)
```

Some of the metadata is correct and some is incorrect (when compared to the raw XML).
Correct:
* dimension_order
* size_c
* size_t
* size_x
* size_y
* size_z
* type
* big_endian
* interleaved
* planes
* significant_bits
* tiff_data_blocks

Incorrect:
* pixels id (should be "Pixels:4")
* channels (it is correct that there is only one channel but incorrect as to it's name / id)

Unsure:
* physical_size_x
* physical_size_y

---

If you get around to handling this, great. If not, I may also be able to look more into it when I start work on https://github.com/AllenCellModeling/aicsimageio/issues/140

**Comment by tlambert03** [comment]
Hey @JacksonMaxfield
I'm curious to know more about why it "should be Pixels:4"?  I guess at the moment I'm thinking that this is either a bug in our extraction of XML from the tiff header, or it's a bug in `bfconvert`...

I broke out the xml extraction in https://github.com/tlambert03/ome-types/pull/72/files, and the XML I get (prior to putting it into `model.OME`) seems to agree with what you posted above.  Here's the XML I'm seeing in that tiff (just the image parts)... If this XML is itself not what you expected, let's try to figure that part out first.  (the code in ome-types that extracts the XML is [here](https://github.com/tlambert03/ome-types/blob/1ef77a3c66fc6cd916563fdb9047cb10f6aa8b4c/src/ome_types/__init__.py#L65-L92))


```xml
    <Image ID="Image:0" Name="">
        <AcquisitionDate>2019-05-09T09:49:02.553</AcquisitionDate>
        <ExperimenterRef ID="Experimenter:0"/>
        <InstrumentRef ID="Instrument:0"/>
        <ObjectiveSettings ID="Objective:1" Medium="Air" RefractiveIndex="1.000293"/>
        <StageLabel Name="Scene position #0" X="17500.0" XUnit="µm" Y="17500.0" YUnit="µm" Z="1352.79" ZUnit="µm"/>
        <Pixels BigEndian="false" DimensionOrder="XYCZT" ID="Pixels:0" Interleaved="false" PhysicalSizeX="0.9082107048835328" PhysicalSizeXUnit="µm" PhysicalSizeY="0.9082107048835328" PhysicalSizeYUnit="µm" SignificantBits="14" SizeC="3" SizeT="1" SizeX="7712" SizeY="6184" SizeZ="1" Type="uint16">
            <Channel AcquisitionMode="WideField" Color="16735231" EmissionWavelength="509.0" EmissionWavelengthUnit="nm" ExcitationWavelength="488.0" ExcitationWavelengthUnit="nm" Fluor="EGFP" ID="Channel:0:0" IlluminationType="Epifluorescence" Name="EGFP" SamplesPerPixel="1">
                <DetectorSettings Binning="1x1" ID="Detector:Axiocam506"/>
                <FilterSetRef ID="FilterSet:1"/>
                <LightPath/>
            </Channel>
            <Channel AcquisitionMode="WideField" Color="-16187137" EmissionWavelength="610.0" EmissionWavelengthUnit="nm" ExcitationWavelength="587.0" ExcitationWavelengthUnit="nm" Fluor="mCherry" ID="Channel:0:1" IlluminationType="Epifluorescence" Name="mCher" SamplesPerPixel="1">
                <DetectorSettings Binning="1x1" ID="Detector:Axiocam506"/>
                <FilterSetRef ID="FilterSet:2"/>
                <LightPath/>
            </Channel>
            <Channel AcquisitionMode="WideField" Color="-1" Fluor="TL Phase Gradient" ID="Channel:0:2" IlluminationType="Transmitted" Name="PGC" SamplesPerPixel="1">
                <DetectorSettings Binning="1x1" ID="Detector:Axiocam506"/>
                <FilterSetRef ID="FilterSet:3"/>
                <LightPath/>
            </Channel>
            <TiffData FirstC="0" FirstT="0" FirstZ="0" IFD="0" PlaneCount="1">
                <UUID FileName="pyramid.ome.tiff">urn:uuid:54bab916-61f0-451d-bfbe-251896f608fb</UUID>
            </TiffData>
            <TiffData FirstC="1" FirstT="0" FirstZ="0" IFD="1" PlaneCount="1">
                <UUID FileName="pyramid.ome.tiff">urn:uuid:54bab916-61f0-451d-bfbe-251896f608fb</UUID>
            </TiffData>
            <TiffData FirstC="2" FirstT="0" FirstZ="0" IFD="2" PlaneCount="1">
                <UUID FileName="pyramid.ome.tiff">urn:uuid:54bab916-61f0-451d-bfbe-251896f608fb</UUID>
            </TiffData>
            <Plane DeltaT="15.388000011444092" DeltaTUnit="s" ExposureTime="150.0" ExposureTimeUnit="s" PositionX="17500.0" PositionXUnit="µm" PositionY="17500.0" PositionYUnit="µm" PositionZ="1352.79" PositionZUnit="µm" TheC="0" TheT="0" TheZ="0"/>
            <Plane DeltaT="15.84500002861023" DeltaTUnit="s" ExposureTime="250.0" ExposureTimeUnit="s" PositionX="17500.0" PositionXUnit="µm" PositionY="17500.0" PositionYUnit="µm" PositionZ="1352.79" PositionZUnit="µm" TheC="1" TheT="0" TheZ="0"/>
            <Plane DeltaT="17.248000144958496" DeltaTUnit="s" ExposureTime="1.0" ExposureTimeUnit="s" PositionX="17500.0" PositionXUnit="µm" PositionY="17500.0" PositionYUnit="µm" PositionZ="1352.79" PositionZUnit="µm" TheC="2" TheT="0" TheZ="0"/>
        </Pixels>
    </Image>
    <Image ID="Image:1" Name="label image">
        <AcquisitionDate>2019-05-09T09:49:02.553</AcquisitionDate>
        <ExperimenterRef ID="Experimenter:0"/>
        <InstrumentRef ID="Instrument:0"/>
        <Pixels BigEndian="false" DimensionOrder="XYCZT" ID="Pixels:1" Interleaved="false" PhysicalSizeX="0.9082107048835328" PhysicalSizeXUnit="µm" PhysicalSizeY="0.9082107048835328" PhysicalSizeYUnit="µm" SignificantBits="8" SizeC="1" SizeT="1" SizeX="422" SizeY="2030" SizeZ="1" Type="uint8">
            <Channel ID="Channel:1:0" SamplesPerPixel="1">
                <LightPath/>
            </Channel>
            <TiffData FirstC="0" FirstT="0" FirstZ="0" IFD="3" PlaneCount="1">
                <UUID FileName="pyramid.ome.tiff">urn:uuid:54bab916-61f0-451d-bfbe-251896f608fb</UUID>
            </TiffData>
            <Plane TheC="0" TheT="0" TheZ="0"/>
        </Pixels>
    </Image>
```

**Comment by JacksonMaxfield** [comment]
Woooooooow.

Alright. Big learning day for me.

You are absolutely correct / `ome-types` is returning the correct result.

I hadn't dumped the XML to a file to view it I had only opened up the file in FIJI w/ the XML viewer opened.
![image](https://user-images.githubusercontent.com/17132317/106220915-5dfdd300-6191-11eb-928e-8c4df5b5251f.png)

I didn't know / Is it well known?? that FIJI will edit the metadata prior to opening the file?

Closing this because it is a FIJI "feature"? I guess?
